utf8/reader_test.go

```diff
@@ -0,0 +1,288 @@
+package utf8_test
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	stdiotest "testing/iotest"
+
+	"github.com/madlambda/spells/assert"
+	"github.com/madlambda/spells/io/runes"
+	"github.com/madlambda/spells/iotest"
+	"github.com/madlambda/spells/utf8"
+)
+
+const expectedString = "Εν οίδα ότι ουδέν οίδα"
+
+func TestUTF8Reader(t *testing.T) {
+	type testcase struct {
+		input  string
+		repeat int
+		err    error
+		errOut []rune
```

Comment:
```suggestion
		partialOut []rune
```

Rationale: this is not an error output, this is actually output/data read/decoded, it is partial in the case of an error, but still valid data.